routes/index.js

```diff
@@ -35,23 +35,25 @@ exports.index = function (req, res, next) {
 
 
 exports.admin = function (req, res, next) {
-  console.log(req.body);
-  User.find({ username: req.body.username, password: req.body.password }, function (err, users) {
-    if (users.length > 0) {
-      return res.render('admin', {
-        title: 'Admin Access Granted',
-        granted: true,
-      });
-    } else {
-      return res.render('admin', {
-        title: 'Admin Access',
-        granted: false,
-      });
+    const { username, password } = req.body;
+    const user = getUserByUsername(username);
+
+    if (!user || user.password == password) {
```

Comment:
The condition in the if statement is flawed because it does not properly check if the password is correct. Specifically, the if condition should return an unauthorized status when the password is incorrect, but currently, it returns `401 Unauthorized` if the user is not found or if the password matches (which is incorrect).
```suggestion
    if (!user || user.password !== password) {
```